app/server.go

```diff
@@ -729,36 +731,64 @@ func (s *Server) removeUnlicensedLogTargets(license *model.License) {
 	})
 }
 
-func (s *Server) startInterClusterServices(license *model.License) {
-	if !*s.Config().ExperimentalSettings.EnableSharedChannels {
-		mlog.Debug("Remote Cluster Service disabled via config")
+func (s *Server) startInterClusterServices(license *model.License, app *App) {
+	// 	TODO: use license once one can be genenerated
+	allowRemoteClusterService := true // license != nil && *license.Features.RemoteClusterService
+	allowSharedChannels := true       //license != nil && *license.Features.SharedChannels
+
+	// Remote Cluster service
+
+	// License check
+	if !allowRemoteClusterService {
+		mlog.Debug("License does not have Remote Cluster services enabled")
 		return
 	}
 
-	if !*s.License().Features.SharedChannels {
-		mlog.Debug("License does not have shared channels enabled")
+	// Config check
+	if !*s.Config().ExperimentalSettings.EnableRemoteClusterService {
+		mlog.Debug("Remote Cluster Service disabled via config")
 		return
 	}
 
 	var err error
 
-	// TODO: check remote cluster service license (MM-30838)
-	if *s.Config().ExperimentalSettings.EnableRemoteClusterService && s.Config().FeatureFlags.EnableRemoteClusterService {
-		s.remoteClusterService, err = remotecluster.NewRemoteClusterService(s)
-		if err != nil {
-			mlog.Error("Error initializing Remote Cluster Service", mlog.Err(err))
-			return
-		}
+	s.remoteClusterService, err = remotecluster.NewRemoteClusterService(s)
+	if err != nil {
+		mlog.Error("Error initializing Remote Cluster Service", mlog.Err(err))
+		return
+	}
 
-		if err = s.remoteClusterService.Start(); err != nil {
-			mlog.Error("Error starting Remote Cluster Service", mlog.Err(err))
-			s.remoteClusterService = nil
-			return
-		}
+	if err = s.remoteClusterService.Start(); err != nil {
```

Comment:
If my understanding here is correct, since we are logging the error and return, that means that in case we fail to start the remote cluster server we will proceed with starting the app server. That's all good, but does that mean that the customer cannot use shared channels anymore? Since in order to recover for this "remote cluster bootup failure" they will need to restart the server.

Reply:
Correct, they'll need to address the failure and restart before they can create new shared channels or send/recv content for existing ones.  User would still be able to read/write any shared channels but sync won't happen until error is resolved.

A NOC would have monitoring on the error to alert them to this on startup, and also their Grafana dashboard would show shared channels offline via the metrics collection.